tests/advanced_visibility_test.go

```diff
@@ -540,59 +540,87 @@ func (s *advancedVisibilitySuite) TestListWorkflow_KeywordQuery() {
 	s.NoError(err)
 	s.Len(resp.GetExecutions(), 0)
 
-	if s.isElasticsearchEnabled {
-		// LIKE is supported on Elasticsearch only.
+	// Prefix search
+	listRequest = &workflowservice.ListWorkflowExecutionsRequest{
+		Namespace: s.namespace,
+		PageSize:  defaultPageSize,
+		Query:     `CustomKeywordField STARTS_WITH "justice"`,
+	}
+	resp, err = s.engine.ListWorkflowExecutions(NewContext(), listRequest)
+	s.NoError(err)
+	s.Len(resp.GetExecutions(), 1)
+}
 
-		// LIKE exact match on Keyword (supported)
-		listRequest = &workflowservice.ListWorkflowExecutionsRequest{
-			Namespace: s.namespace,
-			PageSize:  defaultPageSize,
-			Query:     `CustomKeywordField LIKE "%justice for all%"`,
-		}
-		resp, err = s.engine.ListWorkflowExecutions(NewContext(), listRequest)
-		s.NoError(err)
-		s.Len(resp.GetExecutions(), 1)
+func (s *advancedVisibilitySuite) TestListWorkflow_LikeQuery() {
+	if !s.isElasticsearchEnabled {
+		s.T().Skip("This test is only for Elasticsearch")
+	}
 
-		// LIKE %word% on Keyword (not supported)
-		listRequest = &workflowservice.ListWorkflowExecutionsRequest{
-			Namespace: s.namespace,
-			PageSize:  defaultPageSize,
-			Query:     `CustomKeywordField LIKE "%justice%"`,
-		}
-		resp, err = s.engine.ListWorkflowExecutions(NewContext(), listRequest)
-		s.NoError(err)
-		s.Len(resp.GetExecutions(), 0)
+	id := "es-functional-list-workflow-keyword-query-like-test"
+	wt := "es-functional-list-workflow-keyword-query-like-test-type"
+	tl := "es-functional-list-workflow-keyword-query-like-test-taskqueue"
+	request := s.createStartWorkflowExecutionRequest(id, wt, tl)
 
-		// LIKE %chars% on Keyword (not supported)
-		listRequest = &workflowservice.ListWorkflowExecutionsRequest{
-			Namespace: s.namespace,
-			PageSize:  defaultPageSize,
-			Query:     `CustomKeywordField LIKE "%ice%"`,
-		}
-		resp, err = s.engine.ListWorkflowExecutions(NewContext(), listRequest)
-		s.NoError(err)
-		s.Len(resp.GetExecutions(), 0)
+	searchAttr := &commonpb.SearchAttributes{
+		IndexedFields: map[string]*commonpb.Payload{
+			"CustomKeywordField": payload.EncodeString("foo bar"),
+		},
+	}
+	request.SearchAttributes = searchAttr
+	_, err := s.engine.StartWorkflowExecution(NewContext(), request)
+	s.NoError(err)
 
-		// LIKE NOT %chars% on Keyword (not supported)
-		listRequest = &workflowservice.ListWorkflowExecutionsRequest{
-			Namespace: s.namespace,
-			PageSize:  defaultPageSize,
-			Query:     `CustomKeywordField NOT LIKE "%ice%"`,
-		}
-		resp, err = s.engine.ListWorkflowExecutions(NewContext(), listRequest)
+	time.Sleep(waitForESToSettle)
+
+	// LIKE exact match on Keyword (supported)
+	listRequest := &workflowservice.ListWorkflowExecutionsRequest{
+		Namespace: s.namespace,
+		PageSize:  defaultPageSize,
+		Query:     `CustomKeywordField LIKE "%foo bar%"`,
+	}
+	resp, err := s.engine.ListWorkflowExecutions(NewContext(), listRequest)
+	s.NoError(err)
+	s.Len(resp.GetExecutions(), 1)
```

Comment:
Same here. Please ensure we get the expected record, not just any record

Reply:
These are old tests, soon to be deleted.

Reply:
I don't see new tests that verify this functionality works as expected, are you planning to add them?